mapswipe_workers/mapswipe_workers/utils/api_calls.py

```diff
@@ -27,33 +29,74 @@ def geojsonToFeatureCollection(geojson: dict) -> dict:
     return geojson
 
 
-def query_osm(changeset_id: int):
+def chunks(arr, n_objects):
+    return [
+        arr[i * n_objects : (i + 1) * n_objects]
+        for i in range((len(arr) + n_objects - 1) // n_objects)
+    ]
+
+
+def query_osm(changeset_ids: list, changeset_results):
     """Get data from changesetId."""
-    url = OSM_API_LINK + f"changeset/{changeset_id}.json"
+    id_string = ""
+    for id in changeset_ids:
+        id_string += f"{id},"
 
-    response = retry_get(url)
+    id_string = id_string[:-1]
 
+    url = OSM_API_LINK + f"changesets?changesets={id_string}"
+    response = retry_get(url)
     if response.status_code != 200:
         err = f"osm request failed: {response.status_code}"
         logger.warning(f"{err}")
         logger.warning(response.json())
         raise CustomError(err)
-    response = response.json()["elements"][0]
-    return response
+    tree = ElementTree.fromstring(response.content)
+
+    for changeset in tree.iter("changeset"):
+        id = changeset.attrib["id"]
+        username = changeset.attrib["user"]
+        userid = changeset.attrib["uid"]
+        comment = created_by = None
+        for tag in changeset.iter("tag"):
+            if tag.attrib["k"] == "comment":
+                try:
+                    comment = tag.attrib["v"].replace("\n", " ")
+                except AttributeError:
+                    pass
+            if tag.attrib["k"] == "created_by":
+                created_by = tag.attrib["v"]
+
+        changeset_results[int(id)] = {
+            "username": username,
+            "userid": userid,
+            "comment": comment,
+            "created_by": created_by,
+        }
+    return changeset_results
+
+
+def add_to_properties(attribute: str, feature: dict, new_properties: dict):
```

Comment:
it looks as if this function is not used anymore.

Reply:
ye you are right, sneaked back in on merge